Source/Actors/Player.cs

```diff
@@ -1,2406 +1,2437 @@
+
+        if (Save.Instance.Freecam != Save.FreecamMode.Free)
+        {
+            Manager.FreeCamPosition = Position;
+        }
+    }
+
+    public override void Update()
+    {
+
+        if (Save.Instance.Freecam is Save.FreecamMode.Orbit or Save.FreecamMode.Free)
+        {
+            // Freecam rotation
+            if (Input.Mouse.Down(MouseButtons.Left))
+            {
+                //if the mouse moves too much, it means, the Game reset the mouse somewhere
+                //the camera would behave weirdly, if this was not there
+                //TODO: find a way to limit Mouse Movement on Load?
+                if (Manager.MouseDelta.LengthSquared() > 10000) return;
+                Manager.FreeCamRotation.X += Manager.MouseDelta.X * Time.Delta * FreecamRotateSpeed;
+                Manager.FreeCamRotation.Y += Manager.MouseDelta.Y * Time.Delta * FreecamRotateSpeed;
+                Manager.FreeCamRotation.X %= 360.0f * Calc.DegToRad;
+                Manager.FreeCamRotation.Y = Math.Clamp(Manager.FreeCamRotation.Y, -89.9f * Calc.DegToRad, 89.9f * Calc.DegToRad);
+            }
+
+            // Freecam zoom
+            Manager.FreeCamDistance -= Input.Mouse.Wheel.Y * FreecamZoomSpeed;
+        }
+
+        if (Save.Instance.Freecam == Save.FreecamMode.Free)
+        {
+            // Freecam movement
+            var cameraForward = new Vec2(
+                MathF.Sin(Manager.FreeCamRotation.X),
+                MathF.Cos(Manager.FreeCamRotation.X));
+            var cameraRight = new Vec2(
+                MathF.Sin(Manager.FreeCamRotation.X - Calc.HalfPI),
+                MathF.Cos(Manager.FreeCamRotation.X - Calc.HalfPI));
+
+            // if not currently in TAS, proceed with normal movement
+            if (!Manager.Running)
+            {
+                Manager.FreeCamPosition.X -= Controls.Move.Value.Y * cameraForward.X * FreecamMoveSpeed;
+                Manager.FreeCamPosition.Y -= Controls.Move.Value.Y * cameraForward.Y * FreecamMoveSpeed;
+                Manager.FreeCamPosition.X -= Controls.Move.Value.X * cameraRight.X * FreecamMoveSpeed;
+                Manager.FreeCamPosition.Y -= Controls.Move.Value.X * cameraRight.Y * FreecamMoveSpeed;
+
+                if (Input.Keyboard.Down(Keys.Space))
+                    Manager.FreeCamPosition.Z += FreecamMoveSpeed;
+                if (Input.Keyboard.Down(Keys.LeftControl))
+                    Manager.FreeCamPosition.Z -= FreecamMoveSpeed;
+            }
+            else
+            {
+                bool fw = TASControls.FreeCamCameraForwad.Down;
+                bool rg = TASControls.FreeCamCameraRight.Down;
+                bool bw = TASControls.FreeCamCameraBackwards.Down;
+                bool lf = TASControls.FreeCamCameraLeft.Down;
+
+                float yval = 0;
+                yval -= fw ? 1 : 0;
+                yval += bw ? 1 : 0;
+                float xval = 0;
+                xval += rg ? 1 : 0;
+                xval -= lf ? 1 : 0;
```

Comment:
When i said make it a control, i meant: Create a `VirtualStick`, not 4 direction buttons